zipkin-lens/src/components/TracePage/TraceSummary.jsx

```diff
@@ -11,72 +11,183 @@
  * or implied. See the License for the specific language governing permissions and limitations under
  * the License.
  */
-import React, { useState, useCallback } from 'react';
+import React, { useState, useCallback, useMemo } from 'react';
 import Box from '@material-ui/core/Box';
-// import { AutoSizer } from 'react-virtualized';
+import { AutoSizer } from 'react-virtualized';
+import minBy from 'lodash/minBy';
 
 import TraceSummaryHeader from './TraceSummaryHeader';
-import MiniTimeline from '../MiniTimeline';
-import Timeline from '../Timeline';
+import TraceTimeline from './TraceTimeline';
+import TraceTimelineHeader from './TraceTimelineHeader';
+import SpanDetail from './SpanDetail';
 import { detailedTraceSummaryPropTypes } from '../../prop-types';
+import { hasRootSpan } from '../../util/trace';
 
 const propTypes = {
   traceSummary: detailedTraceSummaryPropTypes.isRequired,
 };
 
 const TraceSummary = ({ traceSummary }) => {
-  const [tsRange, setTsRange] = useState({
-    startTs: 0,
-    endTs: traceSummary.duration,
-  });
-
-  const handleStartAndEndTsChange = useCallback((startTs, endTs) => {
-    setTsRange({
-      startTs,
-      endTs,
+  const isRootedTrace = hasRootSpan(traceSummary.spans);
+  const [rootSpanIndex, setRootSpanIndex] = useState(0);
+  const isRerooted = rootSpanIndex !== 0;
+  const [currentSpanIndex, setCurrentSpanIndex] = useState(0);
+  const [childrenHiddenSpanIds, setChildrenHiddenSpanIds] = useState({});
+  const [isSpanDetailOpened, setIsSpanDetailOpened] = useState(true);
+  const traceTimelineWidthPercent = isSpanDetailOpened ? 60 : 100;
+
+  const handleChildrenToggle = useCallback((spanId) => {
+    setChildrenHiddenSpanIds(prevChildrenHiddenSpanIds => ({
+      ...prevChildrenHiddenSpanIds,
+      [spanId]: prevChildrenHiddenSpanIds[spanId] ? undefined : true,
```

Comment:
Maybe stick with `false` instead of `undefined` for readability - setting to `undefined` doesn't remove from the map so there isn't any performance reason to do it and it looks a little odd next to `true`